src/toolkits/object_detection/object_detector.cpp

```diff
@@ -350,6 +350,74 @@ void object_detector::load_version(iarchive& iarc, size_t version) {
   _load_version(iarc, version, *nn_spec_, state, anchor_boxes());
 }
 
+void object_detector::import_from_custom_model(variant_map_type model_data,
+                                               size_t version) {
+  auto it = model_data.find("_model");
+  flex_dict model = variant_get_value<flex_dict>(it->second);
```

Comment:
Since this value is quite large in memory, let's avoid copying it out. Instead, create a `const &` reference to it, import the model weights, and then delete it at the end